src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -72,20 +72,21 @@
             + "Try 'checkstyle --help' for more information.%n");
 
     private static final String USAGE = String.format(Locale.ROOT,
-          "Usage: checkstyle [-dEghjJtTV] [-c=<configurationFile>] [-C=<checkerThreadsNumber>]"
-          + " [-f=<format>]%n"
-          + "                  [-o=<outputPath>] [-p=<propertiesFile>]"
-          + " [-s=<suppressionLineColumnNumber>]%n"
-          + "                  [-w=<tabWidth>] [-W=<treeWalkerThreadsNumber>]"
-          + " [-e=<exclude>]...%n"
-          + "                  [-x=<excludeRegex>]..."
+          "Usage: checkstyle [-dEghjJtTV] [-b=<xpath>] [-c=<configurationFile>]"
+          + " [-C=<checkerThreadsNumber>]%n"
+          + "                  [-f=<format>] [-o=<outputPath>] [-p=<propertiesFile>]%n"
+          + "                  [-s=<suppressionLineColumnNumber>] [-w=<tabWidth>]"
+          + " [-W=<treeWalkerThreadsNumber>]%n"
+          + "                  [-e=<exclude>]... [-x=<excludeRegex>]..."
           + " <files>...%n"
           + "Checkstyle verifies that the specified source code files adhere to the specified"
           + " rules. By default%n"
           + "violations are reported to standard out in plain format. Checkstyle requires a"
           + " configuration XML%n"
           + "file that configures the checks to apply.%n"
           + "      <files>...            One or more source files to verify%n"
+          + "  -b, --tree-branch-for-xpath=<xpath>%n"
```

Comment:
> Show Abstract Syntax Tree(AST) branches that match XPath

Our other options are worded as `Print`. I think this should follow the same suite.

> "  -j, --javadocTree         Print Parse tree of the Javadoc comment%n"
"  -J, --treeWithJavadoc     Print full Abstract Syntax Tree of the file%n"

> should it be --tree-branch ? 
> --tree-branch-for-xpath=<xpath>%n

Taking into account our other option names, what about something like "treesMatchingXpath"?

> branch already imply xpath.

We need some way to differentiate AST branch from XPath branch. They are both branches, just different types.
We use `tree` for AST before, so lets stick with that. To get around `for` we could use `matching`.

Reply:
> I think this should follow the same suite instead of Show.

agree.

> trees-matching-xpath

I read this as several Trees (of different files), as in my mind 1 tree is 1 file, inside tree we have  bunch of branches. 

> from XPath branch.

I do not understand what is "XPath branch.". In my mind it is expression, nobody care about internal structure of xpath.

> treesMatchingXpath

I still think `brach-matching-xpath` or `tree-brach-matching-xpath` is better.

@strkkk  and @pbludov , please share your thoughts on this.

Reply:
I do not think naming here is that important.
I would suggest to keep `branch` word in the name because short version is `-b` and it would be misleading without `branch` in full name

Reply:
We can use any other short version character, to match long name

Reply:
@romani so what are options to choose from?

Reply:
brach-matching-xpath or
tree-brach-matching-xpath or
trees-matching-xpath or
tree-branch or
tree-branch-for-xpath

I am voting for `tree-branch`.